src/ts/main-page/mainFiltersSlide.ts

```diff
@@ -0,0 +1,230 @@
+    _addFilterWork('stock');
+    _addFilterWork('price');
+}
+
+function _addFilterWork(type: string){
+  let fromSlider: HTMLInputElement;
+  let toSlider: HTMLInputElement;
+  let fromInput: HTMLDivElement;
+  let toInput: HTMLDivElement;
+  const values = _findNeedable(type);
+  // const min = values.first || 0;
+  // const max = values.second || 48;
+  const ratio = values.third || 1;
+
+  if (type === 'price'){
+    fromSlider = document.querySelector<HTMLInputElement>('.price__range-input1') as HTMLInputElement;
+    toSlider = document.querySelector<HTMLInputElement>('.price__range-input2') as HTMLInputElement;
+    fromInput = document.querySelector<HTMLDivElement>('.prive__from-data') as HTMLDivElement;
+    toInput = document.querySelector<HTMLDivElement>('.prive__to-data') as HTMLDivElement;
+
+  }else{
+    fromSlider = document.querySelector<HTMLInputElement>('.stock__range-input1') as HTMLInputElement;
+    toSlider = document.querySelector<HTMLInputElement>('.stock__range-input2') as HTMLInputElement;
+    fromInput = document.querySelector<HTMLDivElement>('.stock__from-data') as HTMLDivElement;
+    toInput = document.querySelector<HTMLDivElement>('.stock__to-data') as HTMLDivElement;
+  }
+  
+  _setToggleAccessible(toSlider);
+
+  fromSlider.oninput = () => _controlFromSlider(fromSlider, toSlider, fromInput, ratio, type);
+  toSlider.oninput = () => _controlToSlider(fromSlider, toSlider, toInput, ratio, type);
+  fromInput.oninput = () => _controlFromInput(fromSlider, fromInput, toInput, ratio, type);
+  toInput.oninput = () => _controlToInput(toSlider, fromInput, toInput, ratio, type);
+
+  _setFromQuery(fromInput, toInput, type);
+  _controlFromInput(fromSlider, fromInput, toInput, ratio, type);
+  _controlToInput(toSlider, fromInput, toInput, ratio, type);
+  _filterFiltraite(type, fromInput, toInput);
+}
+
+function _setFromQuery(fromInput: HTMLDivElement, toInput: HTMLDivElement, type: string){
+  const oldSearch = window.location.search.slice(1,);
+  const oldSearchArray = oldSearch.split('&');
+  const ourReg = `${type}=`;
+  const ourFilter = oldSearchArray.find(item => item.includes(type))?.replace(ourReg,'');
+  const firstAndSecond = ourFilter?.split('%E2%86%95');
+
+  if (firstAndSecond !== undefined && type === 'price'){
+    fromInput.textContent = '€' + firstAndSecond[1];
+    toInput.textContent = '€' + firstAndSecond[0];
+  }else if (firstAndSecond !== undefined){
+    fromInput.textContent = firstAndSecond[1];
+    toInput.textContent = firstAndSecond[0];
+  }
+}
+
+function _findNeedable(type: string){
+  let allItems, min, max, ratio;
+
+  if (type === 'price'){
+    allItems = Array.from(document.querySelectorAll('.product-item')) as ourPrice[];
+    if (allItems !== undefined){
+
+      min = Number(allItems.reduce(function (item: ourPrice, min: ourPrice){
+        if (Number(item.getAttribute('price')) < Number(min.getAttribute('price'))){
+          return item;
+        }
+        return min;

```

Comment:
```suggestion
        return Number(item.getAttribute('price')) < Number(min.getAttribute('price')) 
            ? item 
            : min
```
As an alternative example.